### ISSUE TITLE
native-tls implementation

### ISSUE THREAD

**Issue by BratSinot** [programatik29/axum-server#59]
Greetings!

I want to implement version with `native-tls` to this crate, but I don't sure which way is right.
I can just copy-paste `tls_rustls` folder and rewrite it to `native-tls`, but you use `tokio-rustls` which have `tokio-native-tls` with same API, and it will be a lot of same code.
Or, I can change current implementation through cfg flag.

**Comment by programatik29** [comment]
It is a good idea to keep everything separated. Create a new module and copy paste existing implementation if you wish.

**Comment by BratSinot** [comment]
> @BratSinot you go through with it? I'd be interested in using it to, maybe we could upstream your work?

Nope.

**Comment by programatik29** [comment]
#71 is a good example if anyone wants to implement this.

**Comment by anabyv042** [comment]
Hi! I am trying to implement the integration with `native-tls`. I created `tls_native_tls` folder and implemented `NativeTlsAcceptor` in a very similar way to how it was done for `OpenSSLAcceptor` and `RustlsAcceptor`, however I faced some hardships implementing the `AcceptFuture`. More specifically, native-tls [accept](https://docs.rs/tokio-native-tls/latest/tokio_native_tls/struct.TlsAcceptor.html#method.accept) is just an async function, returning `impl Future<Output=Result<TlsStream<S>, Error>>` which doesn't fit well into the existing model (it's hard to put it inside the `Timeout` in `AcceptFuture::AcceptFuture`). 

What would be the right way to overcome this? Sorry if it's a basic question, I'm a bit new to the language and to low-level async in particular. Thanks!
